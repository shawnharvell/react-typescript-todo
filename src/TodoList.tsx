import React from "react";
import { Todo, ToggleComplete } from "./types";
import { TodoListItem } from "./TodoListItem";

export interface TodoListProps {
  todos: Array<Todo>;
  toggleComplete: ToggleComplete;
}

export const TodoList: React.FC<TodoListProps> = ({ todos, toggleComplete }) => {
  return (
    <div className="todo-list-container">
      <ul>
        {todos.map((todo, index) => (
          <TodoListItem
            key={`${todo.text}---${index}`}
            todo={todo}
            toggleComplete={toggleComplete}
          />
        ))}
      </ul>
    </div>
  );
};
