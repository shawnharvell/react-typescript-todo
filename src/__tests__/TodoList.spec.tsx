import React from "react";
import { render } from "@testing-library/react";

import { TodoList, TodoListProps } from "../TodoList";

const inputProps: TodoListProps = {
  todos: [
    { complete: false, text: "item 1" },
    { complete: true, text: "item 2" },
  ],
  toggleComplete: jest.fn(),
};

describe("TodoList", () => {
  it("snapshot test", () => {
    const { baseElement } = render(<TodoList {...inputProps} />);
    expect(baseElement).toMatchSnapshot();
  });
});
