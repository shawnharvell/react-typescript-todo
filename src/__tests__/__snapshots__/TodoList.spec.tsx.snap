// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList snapshot test 1`] = `
<body>
  <div>
    <div
      class="todo-list-container"
    >
      <ul>
        <li>
          <label>
            <input
              type="checkbox"
            />
            item 1
          </label>
        </li>
        <li>
          <label
            class="complete"
          >
            <input
              checked=""
              type="checkbox"
            />
            item 2
          </label>
        </li>
      </ul>
    </div>
  </div>
</body>
`;
